src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb

```diff
@@ -314,6 +338,49 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
         End Function
 
         <Extension()>
+        Friend Function AreSameCustomModifiers([mod] As ImmutableArray(Of CustomModifier), otherMod As ImmutableArray(Of CustomModifier)) As Boolean
+            Dim count As Integer = [mod].Length
+
+            If (count <> otherMod.Length) Then
+                Return False
+            End If
+
+            For i As Integer = 0 To count - 1 Step 1
+                If (Not [mod](i).Equals(otherMod(i))) Then
+                    Return False
+                End If
+            Next
+
+            Return True
+        End Function
+
+        Private Function HasSameTupleNames(t1 As TypeSymbol, t2 As TypeSymbol) As Boolean
+            If Not t1.IsTupleType AndAlso
+                Not t2.IsTupleType Then
+                Return True
+            End If
+
+            Dim t1Names = t1.TupleElementNames
+            Dim t2Names = t2.TupleElementNames
+
+            If t1Names = Nothing AndAlso t2Names = Nothing Then
+                Return True
+            End If
+
+            If t1Names = Nothing OrElse t2Names = Nothing OrElse t1Names.Length <> t2Names.Length Then
+                Return False
+            End If
+
+            For i = 0 To t1Names.Length - 1
+                If Not IdentifierComparison.Equals(t1Names(i), t2Names(i)) Then
+                    Return False
+                End If
+            Next
+
+            Return True
```

Comment:
Length check and `For` can be replaced by `Return t1Names.SequenceEqual(t2Names, AddressOf IdentifierComparison.Equals)`.